Userland/Libraries/LibJS/Runtime/PropertyName.h

```diff
@@ -43,14 +43,20 @@ class PropertyName {
 
     template<Integral T>
     PropertyName(T index)
-        : m_type(Type::Number)
-        , m_number(index)
     {
         // FIXME: Replace this with requires(IsUnsigned<T>)?
         //        Needs changes in various places using `int` (but not actually being in the negative range)
         VERIFY(index >= 0);
-        if constexpr (NumericLimits<T>::max() >= NumericLimits<u32>::max())
-            VERIFY(index < NumericLimits<u32>::max());
+        if constexpr (NumericLimits<T>::max() >= NumericLimits<u32>::max()) {
+            if (index >= NumericLimits<u32>::max()) {
+                m_string = String::number(index);
+                m_type = Type::String;
+                return;
+            }
+        }
+
+        m_type = Type::Number;
+        m_number = index;
```

Comment:
Unsure if this is what we want but it is much easier on the eyes than: `PropertyName::from_value(global_object, Value(index))` everywhere

Reply:
yeah, that's good :+1: